test/ranking.jl

```diff
@@ -1,18 +1,42 @@
 using StatsBase
+using Missings
 using Compat
 using Compat.Test
 
 a = [1.0, 2.0, 2.0, 3.0, 4.0, 4.0, 4.0, 5.0]
 x = [3.0, 1.0, 2.0, 4.0, 4.0, 2.0, 5.0, 4.0]  # x is a permutated version of a
+xm = [3.0, 1.0, 2.0, 4.0, 4.0, 2.0, 5.0, 4.0, missing]
+s = ["c", "a", "b", "d", "d", "b", "e", "d"] # s is a vector of strings ordered like x
 
 @test ordinalrank(a) == [1, 2, 3, 4, 5, 6, 7, 8]
 @test ordinalrank(x) == [4, 1, 2, 5, 6, 3, 8, 7]
+@test isequal(ordinalrank(xm), [4, 1, 2, 5, 6, 3, 8, 7, missing])
+@test all(ismissing, ordinalrank([missing, missing]))
+@test ordinalrank(s) == ordinalrank(x)
+@test ordinalrank(x, rev = true) == ordinalrank(-x)
+@test ordinalrank(x, lt = (x, y) -> isless(y, x)) == ordinalrank(-x)
 
 @test competerank(a) == [1, 2, 2, 4, 5, 5, 5, 8]
 @test competerank(x) == [4, 1, 2, 5, 5, 2, 8, 5]
+@test isequal(competerank(xm), [4, 1, 2, 5, 5, 2, 8, 5, missing])
+@test all(ismissing, competerank([missing, missing]))
```

Comment:
Here too, better check the result using `isequal`, so that you also check the dimensions. Same below.